### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by AtulPant2704**
CSB Link:- https://hy18b2.csb.app/login
- Clicking on Login button, will take user to login screen if he is not logged in and will logout him if he is logged in.
- Add Alerts for errors, will ad toast for them later on.

### CODE REVIEW

File: src/components/Navbar/Navbar.jsx [hunk 1]
```diff
@@ -1,78 +1,101 @@
...
-              <span className="count">0</span>
-            </Link>
+  const openMenuBar = () => {
+    setMenuOpen(true);
+  }
+  const closeMenuBar = () => {
+    setMenuOpen(false);
+  }
+
+  const checkStatus = (userName) => {
```
Comment (VishalPatil18) [inline src/components/Navbar/Navbar.jsx:20]: The name could be better I guess. Maybe something like `checkAction` would be better in my opinion.

File: src/components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -1,78 +1,101 @@
...
+    return userName ? "Logout" : "Login";
+  }
+
+  const logoutHandler = () => {
+    navigate("/");
+    localStorage.removeItem("token");
+    localStorage.removeItem("user");
+    authDispatch({ type: "LOG_OUT" })
```
Comment (VishalPatil18) [inline src/components/Navbar/Navbar.jsx:28]: As logout is a complete word you could make it as `LOGOUT`

File: src/pages/Authentication/SignUp.jsx [hunk 3]
```diff
@@ -1,61 +1,137 @@
...
+        }
+        else if (response.status === 422) {
+          alert("User already exists");
+        }
+        else if (response.status === 500) {
+          alert("Server Error");
+        }
+      }
+      catch (error) {
+        console.log(error);
```
Comment (RahullRawat) [inline src/pages/Authentication/SignUp.jsx]: You can remove this console.log

**Review by mohammed-sas — APPROVED** [review]
1.functionality is work fine
2.context and associated reducer has been coded well
3.code looks clean

File: src/pages/Authentication/Login.jsx [hunk 4]
```diff
@@ -1,36 +1,97 @@
...
+      }
+      else if (response.status === 401) {
+        alert("Wrong Password");
+      }
+      else if (response.status === 500) {
+        alert("Server Error");
+      }
+    }
+    catch (error) {
+      console.log(error);
```
Comment (RahullRawat) [inline src/pages/Authentication/Login.jsx]: You can remove this console.log

**Review by VishalPatil18 — APPROVED** [review]
Just a couple of suggestions. The rest looks quite good. Great work @AtulPant2704 👍

**Review by RahullRawat — APPROVED** [review]
- Functionality is working fine . 
- CLeaner code and good naming convention, that make reviewer job easy

**Pull request merged**